rand_core/src/lib.rs

```diff
@@ -236,6 +236,14 @@ pub trait TryRngCore {
         UnwrapErr(self)
     }
 
+    /// Wrap RNG with the [`UnwrapRef`] wrapper.
+    fn unwrap_mut(&mut self) -> UnwrapRef<'_, Self>
+    where
+        Self: Sized,
```

Comment:
Do we need this bound?

We should be able to support `&mut dyn TryRngCore<Error = X>` for known `X`.

Reply:
No, I forgot to relax the implicit `Sized` restriction in the UnwrapMut